src/scrcmd_battle_castle.c

```diff
@@ -0,0 +1,280 @@
+#include "global.h"
+#include "assert.h"
+#include "bag.h"
+#include "field_player_avatar.h"
+#include "field_system.h"
+#include "fieldmap.h"
+#include "frontier_data.h"
+#include "game_stats.h"
+#include "heap.h"
+#include "mail.h"
+#include "message_format.h"
+#include "party.h"
+#include "player_data.h"
+#include "pokedex_util.h"
+#include "save_special_ribbons.h"
+#include "save.h"
+#include "scrcmd.h"
+#include "scrcmd_9.h"
+#include "scrcmd_battle_castle.h"
+#include "script.h"
+#include "task.h"
+#include "unk_02030A98.h"
+#include "unk_02035900.h"
+#include "unk_0203DFA4.h"
+#include "unk_0203E348.h"
+#include "unk_0205BB1C.h"
+#include "unk_0205BFF0.h"
+#include "unk_02078E30.h"
+#include "unk_02088288.h"
+#include "unk_02091564.h"
+
+typedef enum BattleCastleChallengeType {
```

Comment:
again, should be in headers, if this is the same as the arcade, possibly merging the two definitions?